### PR TITLE
ssh: add username check to pubkey authentication

### PR DESCRIPTION
**Pull request by stefaus**
Motivation:

At the moment there is no check if a username belongs to a public key in
authorized_keys2 file, which leads to a possible impersonation fraud.
More information in Issue #3425.

Modification:

Changed Ssh2Admin#AdminPublickeyAuthenticator to check if used public key
and username in key file match.
Additionally add a reason to the logged authentication event in case
mapping fails or host check fails.

Result:

Username and key must match, otherwise authentication fails and get
logged with information about failure to domain.access file.

Target: master
Require-notes: yes
Require-book: yes

Signed-off-by: Stefan Moll <stefan@stefaus.de>

### CODE REVIEW

**Comment by paulmillar** [comment]
ok to test

**Comment by paulmillar** [comment]
retest this please

File: modules/dcache/src/main/java/org/dcache/services/ssh2/Ssh2Admin.java [hunk 1]
```diff
@@ -69,6 +70,12 @@
     private static final Logger _log = LoggerFactory.getLogger(Ssh2Admin.class);
     private static final Logger _accessLog =
             LoggerFactory.getLogger("org.dcache.access.ssh2");
+
+    // get the user part from a public key line like
+    // ... ssh-rsa AAAA...... user@hostname
```
Comment (kofemann) [inline modules/dcache/src/main/java/org/dcache/services/ssh2/Ssh2Admin.java]: I would do '@hostname'

**Comment by kofemann** [comment]
Well done, Thanks!

**Pull request merged**